jmespath-core/src/main/java/io/burt/jmespath/Adapter.java

```diff
@@ -82,6 +94,12 @@
   T getProperty(T value, String name);
 
   /**
+   * Returns all of the property names of the given object, or an empty list
+   * when the given value is not an object.
+   */
+  Collection<String> getPropertyNames(T value);
```

Comment:
The adapters are meant to adapt the `T` for the interpreter, and the idea right now is to add methods that help the interpreter, not things that are generic work-with-JSON helpers. When the interpreter is done my intention is to go through the `Adapter` interface and see how it can be minimized, and perhaps that will take it in a more generic direction, but more likely it will take it in a less generic direction.